Python/ceval_macros.h

```diff
@@ -318,6 +310,18 @@ GETITEM(PyObject *v, Py_ssize_t i) {
     do { \
         (COUNTER) = pause_backoff_counter((COUNTER)); \
     } while (0);
+
+#ifdef ENABLE_SPECIALIZATION_FT
+/* Multiple threads may execute these concurrently if thread-local bytecode is
+ * disabled and they all execute the main copy of the bytecode. Specialization
+ * is disabled in that case so the value is unused, but the RMW cycle should be
+ * free of data races.
+ */
+#define RECORD_BRANCH_TAKEN(bitset, flag) \
+    FT_ATOMIC_STORE_UINT16_RELAXED(       \
+        bitset, (FT_ATOMIC_LOAD_UINT16_RELAXED(bitset) << 1) | (flag))
+#else
+#define RECORD_BRANCH_TAKEN(bitset, flag)
```

Comment:
> Or, why have that macro at all, since it's always set to 1?

It's a temporary measure to allow us to gradually enable specialization in free-threaded builds by replacing uses of `ENABLE_SPECIALIZATION` (set to `0` in free-threaded builds) with `ENABLE_SPECIALIZATION_FT` (set to `1` in free-threaded builds and `ENABLE_SPECIALIZATION` in default builds). Eventually we'll set `ENABLE_SPECIALIZATION` to `1` in free-threaded builds too and replace all of the uses `ENABLE_SPECIALIZATION_FT` with `ENABLE_SPECIALIZATION`.

Reply:
I still don't quite get why this should expand to empty code if it's not set, but I guess it doesn't matter if it's a temporary measure anyway :)